README.md

```diff
@@ -22,6 +22,8 @@ The available configuration options are detailed below.
 |:--------|:-----------|
 | StatusHost | IP address or DNS of the NGINX+ instance to retrieve status information from. Defaults to `localhost`. |
 | StatusPort | Port the NGINX+ status endpoint can be reached at. Defaults to `8080`. |
+| APIVersion | API version to use for fetching data. Explicitly set this to `0` for the `legacy` versions of NGINX+. Defaults to `3`, which is for the `newer` API Versions. |
```

Comment:
"newer" and "legacy" have no meaning after your recent changes:

```suggestion
| APIVersion | API version to use for fetching data. Explicitly set this to `0` for the legacy versions of NGINX+. Defaults to the versioned API of `3`. |
```

However, we should try to detect the version (including legacy) by default like was done earlier imo.  Defaulting to 3 for versioned cases makes sense to me.  Otherwise this is a breaking update for legacy API users.

Reply:
We brainstormed upon the point that you made above, and yes, providing compatibility with existing legacy API users is a better option. Following that aspect, we have made the below changes in the logical workflow of the code.

1. Both the APIVversion and APIBasePath are optional.
2. If APIVersion is provided, it'll be considered as versioned API NGINX+. Based on the provided APIBasePath (defaults to '/api' for versioned API) the further code workflow will be executed. If the legacy API is detected (on the server) in this case when APIVersion is provided by the end-user, then the code will exit with a proper error message asking the user to remove the APIVersion from the input configuration.
3. If APIVersion is not provided, the code will try to detect the NGINX+ API (versioned or legacy) on the server. If legacy API is detected, based on the provided APIBasePath (defaults to '/status' for legacy API) the further code workflow will be executed. If versioned API is detected, based on the provided APIBasePath (defaults to '/api' for versioned API) the further code workflow will be executed by taking the default APIVersion of 3.
I have added the same in the readme and also updated the test cases accordingly.
Does this sound reasonable?